linkerd/metrics/src/gauge.rs

```diff
@@ -1,50 +1,44 @@
-use std::fmt::{self, Display};
-
-use tracing::warn;
-
 use super::{FmtLabels, FmtMetric};
+use std::fmt::{self, Display};
+use std::sync::atomic::{AtomicU64, Ordering};
 
 /// An instaneous metric value.
-#[derive(Copy, Clone, Debug, Default, Eq, PartialEq)]
-pub struct Gauge(u64);
+#[derive(Debug, Default)]
+pub struct Gauge(AtomicU64);
```

Comment:
Hmm, for prometheus compatibility, do gauges also need to implement 52-bit precision? I know they don't _currently_...